README.md

```diff
@@ -109,22 +81,118 @@ These can be downloaded from
    curl -LO $(curl -s https://api.github.com/repos/meltwater/awsudo/releases/latest | grep -Eo 'https://github\.com/meltwater/awsudo/releases/download/v.*\.rpm')
    ```
 
-### Example usages
+> **Warning:** You must install Node.js separately, because these packages are not
+> marked as dependent on Node.js within the Debian or Red Hat ecosystems. This
+> facilitates portability across distributions and better accomodates the
+> multitude of ways Node.js can be installed (e.g. using nvm).
+
+#### Docker
+
+awsudo can also be used from its official Docker image, which packages it along
+with its dependencies and the AWS cli.
+
+```bash
+docker pull awsudo/awsudo
+```
+
+> **Note:** See Docker under Usage for details of what the image includes.
 
-#### Command
+## Usage
 
-Basic usage when awsudo is on the PATH:
+Basic usage when awsudo is on the PATH resembles this example:
 
 ```bash
 awsudo arn:aws:iam::123456789012:role/S3Access aws s3 cp ./some/directory s3://some-bucket
 ```
 
-when using with Docker as a command (i.e. not within the container):
+**💡 Tip: awsudo shell!**
+
+When running multiple commands as the same role (especially when using MFA),
+it can be convenient to temporarily authenticate all commands as that role by
+launching a new shell using awsudo:
 
 ```bash
-docker run -v ~/.aws:/root/.aws awsudo/awsudo awsudo arn:aws:iam::123456789012:role/S3Access aws s3 cp ./some/directory s3://some-bucket
+awsudo arn:aws:iam::123456789012:role/S3Access /bin/bash
+aws s3 cp ./some/directory s3://some-bucket
+aws s3 cp ./another/directory s3://some-bucket
+aws s3api list-objects --bucket some-bucket
 ```
 
+> **Note:** the lifespan of the authentication within the shell is dictated by
+> the `--duration` argument to awsudo
+
+### Docker
+
+The Docker image can be used as a direct command:
+
+```bash
+docker run awsudo/awsudo awsudo --help
+```
+
+In order to assume roles, the AWS configuration needs to be mounted as a
+volume in the container:
+
+```bash
+docker run -v ~/.aws:/root/.aws \
+    awsudo/awsudo \
+    awsudo arn:aws:iam::123456789012:role/S3Access aws s3 ls
+```
+
+If you need to operate on local files, those need to be mounted to a working
+directory for the container as well:
+
+```bash
+docker run -v ~/.aws:/root/.aws awsudo/awsudo \
+    -v $PWD:/work -w /work \
```

Comment:
It may help to clarify if this is a bit more "on the nose"

```suggestion
    -v $PWD:/local-working-directort -w /docker-working-directory \
```

Reply:
Great suggestion! Very clear!